### PR TITLE
bugfix: python's api close was non-functional

### PR DESCRIPTION
**Pull request by hackaugusto**
This fixes the python api close, add a regression test for that problem
and add general tests for it.

### CODE REVIEW

**Comment by hackaugusto** [comment]
this supersedes #434 and fixes #1138

**Comment by LefterisJP** [comment]
I am reviewing this one

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,19 +2,20 @@
 Changelog
 =========
 
-* :feature:`1097` Added `--gas-price` command line option
-* :feature:`1038` Introduce an upper limit for the ``settle_timeout`` attribute of the netting channel
-* :bug:`1044` Rename ``/connection`` API endpoint to ``/connections`` for consistency
+* :bug:`1138` REST and Python API close did not working if a transfer was made.
```
Comment (LefterisJP) [inline docs/changelog.rst]: `working` ->`work`

File: docs/changelog.rst [hunk 2]
```diff
@@ -2,19 +2,20 @@
...
+* :feature:`1010` Add ``amount`` and ``target`` to ``EventTransferSentSuccess`` event.
 * :feature:`1022` Include an ``errors`` field in all unsuccessful API responses.
 * :bug:`450` Removed ``block_number`` from contracts events, using block_number from block on which it was mined.
-* :bug:`870` User selectable NAT traversal
+* :bug:`870` User selectable NAT traversal.
 * :feature:`921` Add ``/api/1/connection`` API endpoint returning information about all connected token networks.
 * :bug:`1011` Remove ``settled`` attribute from the NettingChannel smart contract.
 
-* :release:`0.1.0 <2017-09-12>`
+* :release:`0.1.0 <2017-09-12>`.
```
Comment (LefterisJP) [inline docs/changelog.rst]: I think this appears as a section title in the changelog so it should not need a full stop at the end.

File: raiden/tests/api/test_pythonapi.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    assert channel12.state == CHANNEL_STATE_OPENED
+
+    api1.deposit(token_address, api2.address, deposit)
+
+    assert channel12.state == CHANNEL_STATE_OPENED
+    assert channel12.balance == deposit
+    assert channel12.contract_balance == deposit
+    assert api1.get_channel_list(token_address, api2.address) == [channel12]
+
+    # there is a channel open, they must be checking each other
```
Comment (LefterisJP) [inline raiden/tests/api/test_pythonapi.py]: `checking -> healthchecking`  <--- being kind of pedantic here but anyway ..

File: raiden/tests/api/test_pythonapi_regression.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    node1, node2 = raiden_network
+    token_address = token_addresses[0]
+
+    api1 = RaidenAPI(node1.raiden)
+    api2 = RaidenAPI(node2.raiden)
+
+    amount = 10
+    assert api1.transfer(token_address, amount, api2.address)
+
+    api1.close(token_address, api2.address)
```
Comment (LefterisJP) [inline raiden/tests/api/test_pythonapi_regression.py]: as we discussed in the chat after you suggestions, add here more asserts after the close for the invariants of the channel state after `close()`

**Comment by LefterisJP** [comment]
PR is approved. Merge as soon as tests pass.

**Pull request merged**